oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoder.java

```diff
@@ -545,6 +547,21 @@ public void close() {
 
 	}
 
+	/**
+	 * A RestTemplate with timeouts configured to avoid blocking indefinitely when
+	 * fetching JWK Sets while holding the reentrantLock.
+	 */
+	private static final class  RestTemplateWithTimeouts extends RestTemplate {
```

Comment:
To make the intent of this class clearer, will you please name it `RestTemplateWithNimbusDefaultTimeouts`.

Please also remove the extra whitespace between `class` and the class name.